src/prefect/infrastructure/process.py

```diff
@@ -185,7 +206,7 @@ async def kill(self, infrastructure_pid: str, grace_seconds: int = 30):
 
             try:
```

Comment:
I get a CodeQL warning here, but I got the same warning before making any changes so I left this code in the same place I found it aside from changing the exception type.